### ISSUE TITLE
MatTable: Items per page selector does not work properly when choosing *

### ISSUE THREAD

**Issue by SeppPenner** [SamProf/MatBlazor#170]
A simple setup:

```razor
<MatTable Items="@billingElements" class="mat-elevation-z5">
    <MatTableHeader>
        <th scope="col">Date</th>
        <th scope="col">Billing</th>
    </MatTableHeader>
    <MatTableRow>
        <td>@context.Date</td>
        <td>
            <a href="#" id="@GetId(context)" class="btn btn-primary">@context.Description</a>
        </td>
    </MatTableRow>
</MatTable>

@code {
    [Parameter]
    public List<BillingElement> billingElements { get; set; }

    private string GetId(BillingElement billingElement)
    {
        return $"billing_{billingElement.UnixTimeStamp}";
    }
	
	public class BillingElement
    {
        public DateTimeOffset Date { get; set; }

        public string UnixTimeStamp { get; set; }

        public string Description { get; set; }
    }
}
```

This is rendered to:

![image](https://user-images.githubusercontent.com/9639361/62449597-d2a5ac00-b76a-11e9-82b1-afd9fa0de2c3.png)

The * operator shows `1 of -2` pages when only data is available for one page.

I'm using version 1.3.0 in a server side Blazor project.

**Comment by dga711** [comment]
Cannot reproduce in 1.6.2
Please check if this still happens to you.

**Comment by SeppPenner** [comment]
I will try it and report back to you.

**Comment by SeppPenner** [comment]
It still occurs. I will try to create an example project.

**Comment by SeppPenner** [comment]
@dga711 Sorry for the confusion. My code example was wrong. Please use this code:

```csharp
<MatTable Items="@billingElements" class="mat-elevation-z5">
    <MatTableHeader>
        <th scope="col">Date</th>
        <th scope="col">Billing</th>
    </MatTableHeader>
    <MatTableRow>
        <td>@context.Date</td>
        <td>
            <a href="#" id="@GetId(context)" class="btn btn-primary">@context.Description</a>
        </td>
    </MatTableRow>
</MatTable>

@code {
    public List<BillingElement> billingElements { get; set; } = new List<BillingElement> {
        new BillingElement { Date = DateTimeOffset.Now, Description = "Hello", UnixTimeStamp = "12345678"},
        new BillingElement { Date = DateTimeOffset.Now.AddDays(-30), Description = "Hello2", UnixTimeStamp = "12456757"}
    };

    private string GetId(BillingElement billingElement)
    {
        return $"billing_{billingElement.UnixTimeStamp}";
    }

    public class BillingElement
    {
        public DateTimeOffset Date { get; set; }

        public string UnixTimeStamp { get; set; }

        public string Description { get; set; }
    }
}
```

or the example under https://github.com/SeppPenner/MatBlazorTestTable.

**Comment by dga711** [comment]
#202 should fix it when pulled

**Comment by SeppPenner** [comment]
Yes, looks good to me. I will check this with the next version when there is one ;)

**Comment by SeppPenner** [comment]
I can confirm that this is fixed in 1.6.3. Thanks @dga711 for your pull request :)
